### PR TITLE
Handle CSP violation errors when loading images.

### PR DESCRIPTION
**Pull request by carolhmj**
Previously, whenever a texture fetch failed, we would try to load the fallback texture with "data:". If the user's page has a CSP rule that prohibits this, this fallback load would fail and prompt another fallback load, leading to an infinite loop. Now, we stop trying to load the fallback if we hit a security violation error (which is where CSP falls).

This behavior can be reproduced on the Playground by adding the CSP tag to the document:
https://playground.babylonjs.com/#YCY2IL#1105
https://playground.babylonjs.com/#8LFTCH#1452

### CODE REVIEW

File: packages/dev/core/src/Engines/thinEngine.ts
Comment (sebavan) [inline packages/dev/core/src/Engines/thinEngine.ts:3937]: Why not in this case always place UseFallbackTexture to false instead of adding an extra param ?

File: packages/dev/core/src/Misc/fileTools.ts
Comment (RaananW) [inline packages/dev/core/src/Misc/fileTools.ts:232]: Just to stay safe, after removing the event listeners, you should reset the array. Or maybe instead of forEach you can pop each handler and remove it.

File: packages/dev/core/src/Engines/thinEngine.ts
Comment (carolhmj) [reply packages/dev/core/src/Engines/thinEngine.ts:3937]: This would affect all errors, not only the CSP ones so I didn't want to change it

File: packages/dev/core/src/Misc/fileTools.ts
Comment (RaananW) [inline packages/dev/core/src/Misc/fileTools.ts:235]:
```suggestion
        handlersList.length = 0;
```

Avoid garbage collection

**Review by RaananW — APPROVED** [review]
LGTM. Just formatting and waiting for @sebavan final approval

File: packages/dev/core/src/Engines/thinEngine.ts
Comment (sebavan) [reply packages/dev/core/src/Engines/thinEngine.ts:3937]: But in case of CSP error, at least once, the fallback would anyway not work for any errors ?

**Review by sebavan — APPROVED** [review]

**Pull request merged**
